src/Admin/TimelineMeasureAdmin.php

```diff
@@ -53,23 +57,61 @@ protected function configureFormFields(FormMapper $formMapper)
                 ])
             ->end()
         ;
+
+        $this->removeEmptyTranslationsOnSubmit($formMapper);
     }
 
     protected function configureDatagridFilters(DatagridMapper $datagridMapper)
     {
         $datagridMapper
-            ->add('title', null, [
+            ->add('title', CallbackFilter::class, [
                 'label' => 'Titre',
                 'show_filter' => true,
+                'field_type' => TextType::class,
+                'callback' => function (ProxyQuery $qb, string $alias, string $field, array $value) {
+                    if (!$value['value']) {
+                        return;
+                    }
+
+                    $qb
+                        ->join("$alias.translations", 'translations')
+                        ->andWhere('translations.title LIKE :title')
+                        ->setParameter('title', '%'.$value['value'].'%')
+                    ;
+
+                    return true;
+                },
             ])
             ->add('profiles', null, [
                 'label' => 'Profils',
                 'show_filter' => true,
-            ], null, ['multiple' => true])
+            ], null, [
+                'multiple' => true,
+                'query_builder' => function (ProfileRepository $repository) {
+                    return $repository
+                        ->createQueryBuilder('profile')
+                        ->select('profile, translations')
+                        ->join('profile.translations', 'translations')
+                        ->where('translations.locale = :locale')
+                        ->setParameter('locale', 'fr')
+                    ;
+                },
+            ])
             ->add('themes', null, [
                 'label' => 'Thèmes',
                 'show_filter' => true,
-            ], null, ['multiple' => true])
+            ], null, [
+                'multiple' => true,
+                'query_builder' => function (ThemeRepository $repository) {
+                    return $repository
+                        ->createQueryBuilder('theme')
```

Comment:
You should use a custom method of the repo instead. Since the three are similar I guess you could use trait for them as well.